integrationtests/gitjob/git/git_test.go

```diff
@@ -62,309 +65,373 @@ var (
-			if test.expectedErr == nil {
-				require.NoError(t, err)
-			} else {
-				require.Contains(t, err.Error(), test.expectedErr.Error())
-			}
-			if latestCommit != test.expectedCommit {
-				t.Errorf("latestCommit doesn't match. got %s, expected %s", latestCommit, test.expectedCommit)
-			}
+			latestCommit, err := f.LatestCommit(ctx, gitrepo, client)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring(transport.ErrAuthenticationRequired.Error()))
+			Expect(latestCommit).To(BeEmpty())
 		})
-	}
+	})
 
-}
+	Describe("LatestCommit BasicAuth", func() {
+		var ctx context.Context
 
-func TestLatestCommit_BasicAuth(t *testing.T) {
-	ctlr := gomock.NewController(t)
-	defer ctlr.Finish()
-	ctx := context.Background()
+		BeforeEach(func() {
+			ctx = context.Background()
+		})
 
-	tests := map[string]struct {
-		gitrepo        *v1alpha1.GitRepo
-		expectedCommit string
-		expectedErr    error
-	}{
-		"public repo": {
-			gitrepo: &v1alpha1.GitRepo{
+		createBasicAuthSecret := func() *v1.Secret {
+			return &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      config.DefaultGitCredentialsSecretName,
+					Namespace: "",
+				},
+				Data: map[string][]byte{v1.BasicAuthUsernameKey: []byte(gogsUser), v1.BasicAuthPasswordKey: []byte(gogsPass)},
+				Type: v1.SecretTypeBasicAuth,
+			}
+		}
+
+		It("public repo", func() {
+			gitrepo := &v1alpha1.GitRepo{
 				Spec: v1alpha1.GitRepoSpec{
 					Repo:   url + "/test/public-repo",
 					Branch: "master",
 				},
-			},
-			expectedCommit: latestCommitPublicRepo,
-			expectedErr:    nil,
-		},
-		"private repo": {
-			gitrepo: &v1alpha1.GitRepo{
+			}
+
+			secret := createBasicAuthSecret()
+			f := fleetgit.Fetch{}
+			client := fake.NewClientBuilder().WithRuntimeObjects(secret).Build()
+			latestCommit, err := f.LatestCommit(ctx, gitrepo, client)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(latestCommit).To(Equal(latestCommitPublicRepo))
+		})
+
+		It("private repo", func() {
+			gitrepo := &v1alpha1.GitRepo{
 				Spec: v1alpha1.GitRepoSpec{
 					Repo:   url + "/test/private-repo",
 					Branch: "master",
 				},
-			},
-			expectedCommit: latestCommitPrivateRepo,
-			expectedErr:    nil,
-		},
-	}
+			}
+
+			secret := createBasicAuthSecret()
+			f := fleetgit.Fetch{}
+			client := fake.NewClientBuilder().WithRuntimeObjects(secret).Build()
+			latestCommit, err := f.LatestCommit(ctx, gitrepo, client)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(latestCommit).To(Equal(latestCommitPrivateRepo))
+		})
+	})
+
+	Describe("LatestCommit SSH", func() {
+		var privateKey string
+		var sshPort string
 
-	for name, test := range tests {
-		t.Run(name, func(t *testing.T) {
+		BeforeEach(func() {
+			ctx := context.Background()
+			port, err := container.MappedPort(ctx, "22")
+			Expect(err).NotTo(HaveOccurred())
+			sshPort = port.Port()
+		})
+
+		// Set up SSH key once per describe block
+		BeforeEach(func() {
```

Comment:
With a `BeforeEach` block, SSH keys will be set up before running each test case.
The behaviour is correct with the check on empty private keys below, but the comment is misleading ;)